src/tools/kdb/get.c

```diff
@@ -0,0 +1,229 @@
+	ksAppendKey (spec, keyNew (COMMAND_SPEC_KEY (COMMAND_NAME) "/all", KEY_META, "description", "Consider all of the keys", KEY_META,
+				   "opt", "a", KEY_META, "opt/long", "all", KEY_META, "opt/arg", "none", KEY_END));
+	ksAppendKey (spec, keyNew (COMMAND_SPEC_KEY (COMMAND_NAME) "/name", KEY_META, "description", "The name of the key", KEY_META,
+				   "args", "indexed", KEY_META, "args/index", "0", KEY_END));
+
+	ADD_BASIC_OPTIONS (spec, COMMAND_SPEC_KEY (COMMAND_NAME))
+}
+
+int execGet (KeySet * options, Key * errorKey)
+{
+	int ret = 0;
+	bool verbose = false;
+	elektraKeyToBoolean (GET_OPTION_KEY (options, "verbose"), &verbose);
+
+	bool all = false;
+	elektraKeyToBoolean (GET_OPTION_KEY (options, "all"), &all);
+
+	const char * rawName = GET_OPTION (options, "name");
+	bool resolved = false;
+	char * name = expandKeyName (options, rawName, &resolved);
+	if (name == NULL)
+	{
+		ELEKTRA_SET_VALIDATION_SEMANTIC_ERRORF (errorKey, "could not resolve bookmark in '%s'", rawName);
+		return -1;
+	}
+	if (verbose && resolved)
+	{
+		printf ("resolved bookmark: \'%s\' -> \'%s\'\n", rawName, name);
+	}
+
+	KeySet * searchIn = ksNew (0, KS_END);
+	KDB * handle = kdbOpen (NULL, errorKey);
+	Key * toLookUp = keyNew (name, KEY_END);
+	Key * parentKey = toLookUp;
+	if (all)
+	{
+		parentKey = keyNew ("/", KEY_END);
+	}
+	if (kdbGet (handle, searchIn, parentKey) == -1)
+	{
+		ELEKTRA_SET_VALIDATION_SEMANTIC_ERRORF (errorKey, "could not load '%s': %s", name, GET_ERR (toLookUp));
+		kdbClose (handle, errorKey);
+		ksDel (searchIn);
+		keyDel (toLookUp);
+		keyDel (parentKey);
+		return -1;
+	}
+	kdbClose (handle, errorKey);
+	keyDel (parentKey);
+
+	setCallback (toLookUp, warnOnMeta);
+	if (verbose)
+	{
+		printf ("got %ld keys\n", ksGetSize (searchIn));
+		setCallback (toLookUp, printTrace);
+	}
+
+	Key * found = ksLookup (searchIn, toLookUp, KDB_O_CALLBACK);
+
+	if (keyCopyAllMeta (errorKey, toLookUp) == -1)
+	{
+		ELEKTRA_SET_INTERNAL_ERROR (errorKey, "could not copy key meta to errorKey");
+		ksDel (searchIn);
+		keyDel (toLookUp);
+		return -1;
+	}
+
+	if (found != NULL)
+	{
+		if (verbose)
+		{
+			if (keyGetNamespace (found) == KEY_NS_DEFAULT)
+			{
+				printf ("The key was not found in any other namespace, taking the default\n");
+			}
+			printf ("The resulting keyname is %s\n", keyName (found));
+			printf ("The resulting value size is %ld\n", keyGetValueSize (found));
+		}
+		if (keyIsBinary (found))
+		{
+			ssize_t binSize = keyGetValueSize (found);
+			if (verbose)
+			{
+				printf ("The key is %s.\n", binSize == 0 ? "null" : "binary");
+			}
+			char * buffer = elektraMalloc (binSize);
+			if (buffer == NULL)
+			{
+				ELEKTRA_SET_RESOURCE_ERROR (errorKey, "could not allocate memory for key bin value");
+			}
+			else
+			{
+				keyGetBinary (found, buffer, sizeof buffer);
+				for (ssize_t pos = 0; pos < binSize; pos++)
+				{
+					printf ("%x", buffer[pos]);
+				}
+				elektraFree (buffer);
+			}
```

Comment:
```suggestion
			char * buffer = keyValue (found);
			for (ssize_t pos = 0; pos < binSize; pos++)
			{
				printf ("%x", buffer[pos]);
			}
```

No need to copy the value just for printing